peekingduck/pipeline/nodes/input/utils/read.py

```diff
@@ -40,13 +41,21 @@ def __init__(self, input_source: str, mirror_image: bool, buffering: bool) -> No
             if str(input_source).isdigit():
                 # to eliminate opencv's "[WARN] terminating async callback"
                 self.stream = cv2.VideoCapture(input_source, cv2.CAP_DSHOW)
+            elif Path(input_source.lower()).suffix == ".png":
+                self.stream = PNGReader(input_source)
             else:
                 # no cv2.CAP_DSHOW flag if input_source is file
                 self.stream = cv2.VideoCapture(str(input_source))
         else:
-            self.stream = cv2.VideoCapture(
-                str(input_source) if isinstance(input_source, Path) else input_source
-            )
+            # Linux or macOS
+            if Path(input_source.lower()).suffix == ".png":
+                self.stream = PNGReader(input_source)
+            else:
+                self.stream = cv2.VideoCapture(
+                    str(input_source)
+                    if isinstance(input_source, Path)
+                    else input_source
```

Comment:
I'm seeing multiple instances of manually casting `input_source` to `Path`, will `input_source` ever be a `Path` type? Especially when type hinting in `__init__` says it's a `str`. If not, then is this conditional expression necessary?

If it's not easy to determine it now, we can KIV this for when we properly redo type hinting.